idea/idea-frontend-fir/idea-fir-low-level-api/src/org/jetbrains/kotlin/idea/fir/low/level/api/lazy/resolve/FirLazyDeclarationResolver.kt

```diff
@@ -56,7 +53,9 @@ internal class FirLazyDeclarationResolver(
         val firFile = declaration.getContainingFile()
             ?: error("FirFile was not found for\n${declaration.render()}")
         val provider = firFile.session.firIdeProvider
-        val fromPhase = if (reresolveFile) declaration.resolvePhase else minOf(firFile.resolvePhase, declaration.resolvePhase)
+        val fromPhase: FirResolvePhase by lazy {
```

Comment:
We do not need thread safe lazy here :) let's use one with `LazyThreadSafetyMode.NONE`

Reply:
Also, let's add some comment to the code describing why this variable is lazy